WordPress/Helpers/WPHookHelper.php

```diff
@@ -0,0 +1,114 @@
+<?php
+/**
+ * WordPress Coding Standard.
+ *
+ * @package WPCS\WordPressCodingStandards
+ * @link    https://github.com/WordPress/WordPress-Coding-Standards
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+namespace WordPressCS\WordPress\Helpers;
+
+use PHPCSUtils\Utils\PassedParameters;
+
+/**
+ * Helper utilities for recognizing functions related to the WP Hook mechanism.
+ *
+ * @package WPCS\WordPressCodingStandards
+ * @since   3.0.0 The property in this class was previously contained in the
+ *                `WordPressCS\WordPress\Sniff` class and has been moved here.
+ */
+final class WPHookHelper {
+
+	/**
+	 * A list of functions that invoke WP hooks (filters/actions).
+	 *
+	 * @since 0.10.0
+	 * @since 0.11.0 Changed from public static to protected non-static.
+	 * @since 3.0.0  - Moved from the Sniff class to this class.
+	 *               - The property visibility has changed from `protected` to `private static`.
+	 *                 Use the `get_function_names()` method for access.
+	 *               - The format of the value has changed from a non-relevant boolean to
+	 *                 an array with the parameter position and name(s) for the hook name parameter.
+	 *
+	 * @var array<string, <string, int|string|array>> Function name as key, array with target
```

Comment:
> The possibility of `name` being an array isn't represented in the rest of the DocBlock, the current values, or in its usage.

But it should still be mentioned to prevent confusion later. The `name` key can have either a string or an array as value to allow for parameters which have been renamed since the release of PHP 8.0,. And as WP still doesn't officially support named parameters, the chances of a parameter rename happening at some point in the future is realistic.

Also, on the next line , it says "parameter position and name(s) as value.", same as in the changelog entry, so the possibility of `name` being an array entry is loosely mentioned already in the DocBlock.